### PR TITLE
[MRG+1] DOC Convert axes docstrings to numpydoc: #7205

### PR DESCRIPTION
**Pull request by cfackler**
I've converted several docstrings for the axes class to numpydoc format.

### CODE REVIEW

File: lib/matplotlib/axes/_base.py [hunk 1]
```diff
@@ -2783,49 +2783,77 @@ def set_xbound(self, lower=None, upper=None):
...
+            The left and right xlims may be passed as the tuple
+            (`left`, `right`) as the first positional argument (or as
+            the `left` keyword argument).
 
-          *emit*: [ *True* | *False* ]
-            Notify observers of limit change
+        Returns
+        -------
+        xlimits : tuple
+            Returns the current x-axis limits, reflecting any changes
```
Comment (NelleV) [inline lib/matplotlib/axes/_base.py]: I'd say "returns the new x-axis limits"

File: lib/matplotlib/axes/_base.py [hunk 2]
```diff
@@ -3037,49 +3065,78 @@ def set_ybound(self, lower=None, upper=None):
...
+            limit unchanged). The previous name `ymin` may be used
+            instead.
 
-          set_ylim((bottom, top))
-          set_ylim(bottom, top)
-          set_ylim(bottom=1) # top unchanged
-          set_ylim(top=1) # bottom unchanged
+        top : scalar, optional
+            The top ylim (default: None, which leaves the top limit
+            unchanged). The previous name `ymax` may be used instead.
```
Comment (NelleV) [inline lib/matplotlib/axes/_base.py]: I'd avoid mentioning the old API. As much as possible, we should advertise one and only way to do an action.

**Comment by cfackler** [comment]
@NelleV Thanks for the suggestions. I agree and have updated the PR.

**Comment by NelleV** [comment]
Thanks @cfackler for the patch!

The cherry-pick to v.2.x conficts and I don't have time to look into it tonight.

**Comment by tacaswell** [comment]
Backported to v2.x as 814fff5

**Pull request merged**